src/solvers/dgsem_unstructured/dg_2d.jl

```diff
@@ -383,12 +384,12 @@ end
 
 # inlined version of the boundary flux calculation along a physical interface where the
 # boundary flux values are set according to a particular `boundary_condition` function
-@inline function calc_boundary_flux!(surface_flux_values, t, boundary_condition,
+@inline function calc_boundary_flux!(surface_flux_values, t, boundary_condition::BC,
                                      mesh::UnstructuredMesh2D,
                                      nonconservative_terms::False, equations,
                                      surface_integral, dg::DG, cache,
                                      node_index, side_index, element_index,
-                                     boundary_index)
+                                     boundary_index) where {BC}
```

Comment:
Is this really necessary? I would have expected the compiler heuristics to work well since `boundary_condition` is called below.

Reply:
No, that is (at least currently) not required.

Reply:
I would avoid it in this case since it's not required and increases the code complexity a bit. What do you think, @sloede?

Reply:
If possible, let's try to avoid it. Every unnecessary type/parameter specification adds to the mental load when trying to understand a section of code 🤓